sdk/tcp_listener.go

```diff
@@ -14,6 +16,14 @@ const (
 )
 
 func newTCPListener(address string, pluginName string) (net.Listener, string, error) {
+	if address == "" {
+		return nil, "", fmt.Errorf("plugin address cannot be empty")
+	}
+	addrParts := strings.SplitN(address, ":", 2)
```

Comment:
My concern is if net.SplitHostPort on Windows will error out because it doesnt understand empty hostnames, which are perfectly valid, which is also the reason for this PR :) I dont have a setup to verify this, atm.

However, yes, we should account for IPv6. I'll try to get a Windows setup to confirm that SplitHostPort works as intended.